src/base/bittorrent/torrenthandle.cpp

```diff
@@ -596,8 +596,7 @@ bool TorrentHandle::removeTag(const QString &tag)
 
 void TorrentHandle::removeAllTags()
 {
-    // QT automatically copies the container in foreach, so it's safe to mutate it.
-    foreach (const QString &tag, m_tags)
+    for (const QString &tag : qAsConst(m_tags))
```

Comment:
I think this form will have problems (crash), better:
```c++
const QSet<QString> tags {m_tags};
for (const QString &tag : tags)
    removeTag(tag);
```

Reply:
or
```c++
for (const QString &tag : tags())
    removeTag(tag);
```

Reply:
Actually i have tested qAsConst and what you said and it doesn't crash, also tested without any of them and naturally it crashed. If you still think i should do it i will.

Reply:
Never mind it crashed when doing it to multiple torrents.